test/test_grid.py

```diff
@@ -497,32 +498,6 @@ class TestConnectivity(TestCase):
     ugrid_filepath_02 = current_path / "meshfiles" / "ugrid" / "outRLL1deg" / "outRLL1deg.ug"
     ugrid_filepath_03 = current_path / "meshfiles" / "ugrid" / "ov_RLL10deg_CSne4" / "ov_RLL10deg_CSne4.ug"
 
-    def test_build_edge_nodes(self):
-        """Tests the construction of (``Mesh2_edge_nodes``) on an MPAS grid
-        with known edge nodes."""
-
-        # grid with known edge node connectivity
-        mpas_grid_xr = xr.open_dataset(self.mpas_filepath)
-        mpas_grid_ux = ux.Grid(mpas_grid_xr)
-        edge_nodes_expected = mpas_grid_ux.ds['Mesh2_edge_nodes'].values
-
-        # arrange edge nodes in the same manner as Grid._build_edge_node_connectivity
-        edge_nodes_expected.sort(axis=1)
-        edge_nodes_expected = np.unique(edge_nodes_expected, axis=0)
-
-        # construct edge nodes
-        mpas_grid_ux._build_edge_node_connectivity()
-        edge_nodes_output = mpas_grid_ux.ds['Mesh2_edge_nodes'].values
-
-        assert np.array_equal(edge_nodes_expected, edge_nodes_output)
-
-        # euler's formula (n_face = n_edges - n_nodes + 2)
-        n_face = mpas_grid_ux.nMesh2_node
-        n_node = mpas_grid_ux.nMesh2_face
-        n_edge = edge_nodes_output.shape[0]
-
-        assert (n_face == n_edge - n_node + 2)
-
```

Comment:
By examining the MPAS read-in process, it can be discovered that the `np.unique` function in `Grid._build_edge_nodes_connectivity` does not actually perform any unique operations. This is because MPAS has its own "edge_nodes_connectivity," which contains only unique edges. Therefore, the returned `edge_nodes_unique` and `inverse_indices` cannot be reshaped into the `face_edges_connectivity` 2D matrix. As a result, the `face_edges_connectivity` cannot be built unless the `edge_nodes_connectivity` and `inverse_indices` are rebuilt from the MPAS `face_nodes_connectivity`. The deleted testcase is essentially a redundant one, as it mirrors the exact same process as `Grid._build_edge_nodes_connectivity` and therefore does not provide any additional value.

Let's say the MPAS data has the `mesh2_face_nodes` as 162 face * 6 nodePer_face
1. MPAS has its own "edge_nodes_connectivity", which are all unique edges (480 unique edges)
2. So in the `Grid._build_edge_nodes_connectivity`, the `np.unique` does nothing
```python
        # unique edge nodes
        edge_nodes_unique, inverse_indices = np.unique(edge_nodes,
                                                       return_inverse=True,
                                                       axis=0)
```
The returned ` edge_nodes_unique` is still `480` and the `inverse_indices` is also 480 (The correct size should be 162*6=972), which means this `inverse_indices` cannot be reshaped into the `face_edges_connectivity` (A 162*6 2D matrix). 

As I mentioned in issue #290, there's no way we can build the `face_edges_connectivity` unless we force rebuilding the `edge_nodes_connectivity` and `inverse_indices` from the MPAS `face_nodes_connectivity`

3. So since the unique didn't do any actual work. So if we summarize the pathway when the MPAS read in in the original `Grid._build_edge_nodes_connectivity` using the deleted testcase, it will be:

- Since `if "Mesh2_edge_nodes" not in self.ds or repopulate:` is false, it will run `edge_nodes = self.ds["Mesh2_edge_nodes"]`
- `edge_nodes.sort(axis=1)`
- `edge_nodes_unique = edge_nodes` This is what actually happened when we called `np.unique()`
- All the following masking and fill_value or duplicate edges treatment is invalidate

4. Now we look back to your deleted test case, again th `np.unique()` will not do any actual work instead of reordering the data:
```python
        # grid with known edge node connectivity
        mpas_grid_xr = xr.open_dataset(self.mpas_filepath)
        mpas_grid_ux = ux.Grid(mpas_grid_xr)
        edge_nodes_expected = mpas_grid_ux.ds['Mesh2_edge_nodes'].values

        # arrange edge nodes in the same manner as Grid._build_edge_node_connectivity
        edge_nodes_expected.sort(axis=1)
        edge_nodes_expected = np.unique(edge_nodes_expected, axis=0)
```
It's identical to what's inside the `Grid._build_edge_nodes_connectivity`.

Reply:
I can see that with the addition of `repopulate` this test case wasn't performing as expected. However, the purpose of this test was to test our algorithm for obtaining the edge nodes with a dataset that _already_ contained edge nodes.

I would like to keep this test, but we would need to add `repopulate=True` to the call. Then, we will still be comparing the output of our algorithm versus the provided edge nodes.

Reply:
`repolulate` actually will not change the original case at all. In fact, the addition of `repopulate` helps.

Previously, we only has this line in `Grid.build_edge_nodes_connectivity`
```python
if "Mesh2_edge_nodes" not in self.ds:
    # Build the connectivity

```
As explained above, since MPAS already has its own `Mesh2_edge_nodes`, it will never go into the actual 
building process and won't build the `inverse_indices`. I also described this phenomenon in issue #290 

By adding the flag `repopulate`, we can manually call the process bypassing the existed `Mesh2_edge_nodes`

But I can call it with`repopulate=True` and compare it with the original ones